### DISCUSSION TITLE
How to generate a set of replicable files and folders in the project?

### DISCUSSION THREAD

**Discussion by mrlucasrib** [cookiecutter/cookiecutter#1673] [Q&A]
I would like to know if it is possible to create sub-templates and if so, how to use them. I have a project where we have a very repetitive job of creating new UseCases, where it is necessary to create a different folder with several files with very similar things between them. I wish there was a command like: `cookiecutter add usecase GetUserData` and then it would create some files in a folder with the same name.
Is this possible? if yes, how?

Also, someone in a specific part of the code (on a router) must add things to an already existing file. is this possible? How?

**Comment by readytheory** [comment]
Did you get this resolved to your satisfaction?  Your main requirement sounds to me like you are asking for exactly what cookiecutter is intended to do.  You give it values for some parameters and it generates code which is similar to many other projects, where all the variation is anticipated in a master template.

The second part -- "must add things to an already existing file" -- I don't think CC is going to do that naturally.  If you mean, the code that is generated is supposed to do that, that would be fine.
